cs3/storage/provider/v1beta1/resources.proto

```diff
@@ -313,9 +316,8 @@ message RecycleItem {
   // The key to identify the deleted resource.
   string key = 3;
   // REQUIRED.
-  // The original path of the deleted resource.
-  // MUST start with the slash character (/).
-  string path = 4;
+  // The original reference of the deleted resource.
+  Reference ref = 4;
```

Comment:
@butonic this change might lead to some inconsistencies. Do we want to return either of resource ID and path here? Won't having both `ResourceId` and `path` fields like in `ResourceInfo` make more sense?

Reply:
Restoring trash is similar to creating directories or uploading files. You actually need to pass a reference for the parent and a name for the file. When listing a directory you get `ResourceInfo` items, which carry the `id` of the resource and a path.

When listing `RecycleItem`s the `key` is what you need to use when making a `RestoreRecycleItem` request. So that is the crucial bit of information. I don't think we need an `id`. If we add it clients might want to stat a file in the trash. While I like the idea of being able to navigate a trashed folder, eg. by using ListContainer with that trashed id ... it might not be possible to implement on every storage driver. Should the CS3 API allow it? I think so. What is your take on this?

But when listing `RecycleItem`s there is no relative anchor. Currently, it is not even possible to limit the listing to a certain storage space.

- [ ] Add `ref:Reference` to `ListRecycleRequest`

With that in place we can apply the same logic to the `RecycleItem` `path` property as we have it for `ResourceInfo`:
- when no `Ref.ResouceID` was given in the request the `RecycleItem` `path` property in the response is absolute and must start with `/`
- when a `Ref.ResouceID` was given in the request the `RecycleItem` `path` property in the response is relative and must start with a `.`

Reply:
> With that in place we can apply the same logic to the RecycleItem path property as we have it for ResourceInfo

LGTM!

> While I like the idea of being able to navigate a trashed folder, eg. by using ListContainer with that trashed id ... it might not be possible to implement on every storage driver. Should the CS3 API allow it? I think so. What is your take on this?

According to me, this can be a pretty nice addition. Of course, some drivers might not implement it which should also be fine. `gateway. ListRecycleRequest` already has a ref parameter. We can use the same request in `storageprovider` but we can extend it later on.

Can you update the comment to reflect @refs's point? I'll merge then

Reply:
Your suggestion sounds good to me. Maybe we can wait for @labkode to take a look as well